pom.xml

```diff
@@ -38,33 +37,40 @@
   </issueManagement>
 
   <properties>
-    <sonar.version>4.5.6</sonar.version>
+    <sonar.version>6.7.1</sonar.version>
     <sonar.pluginName>Clover</sonar.pluginName>
     <sonar.pluginClass>org.sonar.plugins.clover.CloverPlugin</sonar.pluginClass>
   </properties>
 
   <dependencies>
     <dependency>
-      <groupId>org.codehaus.sonar</groupId>
+      <groupId>org.sonarsource.sonarqube</groupId>
       <artifactId>sonar-plugin-api</artifactId>
       <version>${sonar.version}</version>
       <scope>provided</scope>
     </dependency>
 
+    <dependency>
```

Comment:
From my perspective, this pull request is a single unit of work based on the end goal of updating the `Sensor` functionality and passing the Travis build (compile/test/deploy/test). 

Here's the development process I followed:
1) Change the code to use the new `Sensor` API
2) Write/modify tests to verify the behavior
3) Update the build logic so that the plugin compiles, tests, and can be deployed correctly

Additionally, I took the following steps as part of our internal process:
1) Had the changes code-reviewed by a member of my team, and demonstrated the changes in action
2) Worked with internal IT to deploy the resulting plugin to a test instance of Sonar 6.7.1 and verified that coverage reports are processed correctly

If I omitted any of the development steps, there are only two flows that I can think of:
- Work on a branch outside the Travis build flow
- Comment out all the failing / non-compiling code as a temporary measure

Can you please let me know how you want to proceed?

Thanks!

Reply:
@Tony-Proum ,

Can you please let me know if there are any status updates?

Reply:
IMHO, the Build/Deploy logic have to be rewritten in another thread. As SonarSource have left the support for this plugin, we will have to find a place to store/deploy artifacts by our own etc. So we have to think about a lot of think that should not be covered by this thread. Also, I think that improving test coverage on existing behavior should be extracted from this PR in order to keep only the relevant piece of work needed to `Use the new API`. In other word, we should only include point 1 and point 2 and extract point 3 from this PR.

The idea behind splitting this PR is to allows other people to work on each issue in it's own thread this way, multiple issues may be address at the same time.

Reply:
@Tony-Proum ,

A couple of questions:
1. Are you okay with merging in untested code?
1. Does the Travis build need to complete?

Once we have the answers to the above questions, can you please create issues in this backlog that are scoped as you want, and then we can start addressing them in the sequence you want. 

As far as where the code is stored/deployed, is there a reason to not use Maven Central as the codebase already does?

I'd really like to get this moving and completed, so if we have a set of ordered tasks to complete, we can parallelize as you indicated and get this release out.

Thanks!

Reply:
Hello @mlschechter 
I'm not ok with merging untested code. I think that the description/title of an Issue/PR should reflect the content of the changeset created. 
Reviewing this PR, it seems like the content of the changeset is too broad and not focused on the expected feature.

I think that we should first fix the build in order to have a clean workflow for this project (sorry for the delay, we are working on it).

If you need to use this plugin version now, you should be able to build it locally and put the resulting jar in the plugin folder of your sonarqube server. I hope that this workaround may help to cut of the emergency of merging this PR now and would allow us to fix the build before.

I'm ok with this point: I have to create a bunch of issues to reflect what we have to do. I forgot this point for a while, thanks you for pointing this problem :+1:

Reply:
@Tony-Proum ,

My Sonar admins will not deploy an unreleased plugin (even if I did the work). Let's go with the sequential split as we discussed above, and keep things moving 😄